### ISSUE TITLE
Streamlink-twitch-GUI stuck on floating mode on hyprland, maybe any tiling WM since 2.5.x

### ISSUE THREAD

**Issue by Porkepix** [streamlink/streamlink-twitch-gui#1012]
### Checklist

- [X] I understand that **Streamlink Twitch GUI** is just a launcher for **Streamlink**
- [X] [I have read the contribution guidelines](https://github.com/streamlink/streamlink-twitch-gui/blob/master/CONTRIBUTING.md)
- [X] [I have checked the list of open and recently closed issues](https://github.com/streamlink/streamlink-twitch-gui/issues?q=is%3Aissue+sort%3Aupdated-desc)
- [X] [I have checked the commit log of the master branch](https://github.com/streamlink/streamlink-twitch-gui/commits/master)

### Streamlink Twitch GUI version

2.5.1

### Streamlink version

6.7.4

### Operating system, environment and configuration details

ArchLinux, Hyprland

### Description

Since 2.5.X, probably due to nw.js changes, streamlink-twitch-GUI only offers floating or full screen modes at least on Hyprland, maybe also other compositors such as Sway.
That means it gets stuck on foreground in both cases, unless we set another tile as fullscreen to go on top of it. In the end it completely breaks the tiling.
This is a regression from 2.4.x which had no problems (I did a downgrade to check issue is gone on 2.4.x). Gnome gets no issue around this (which makes sense, no tiling by default).

Note that the problem is when it's running with the default parameters, ie. on XWayland.
When forcing the use of Wayland with --ozone-platform=wayland, that issue is gone, but some other issues arise such as the default tray icon getting its own tile window.

Note that Hyprland maintener suggested it could be an issue with window requested max size and suggested to try to override it from Hyprland's side, but it didn't fixed anything. (If the max size is smaller than what the tile size would be, then Hyprland don't tile it).

### Debug log

No relevant logs, but here are some n case it might show something interesting for some issues with some visible warnings/errors:
```
[64168:64168:0515/112954.786929:ERROR:policy_logger.cc(157)] :components/enterprise/browser/controller/chrome_browser_cloud_management_controller.cc(163) Cloud management controller initialization aborted as CBCM is not enabled.
[0515/112954.805744:ERROR:elf_dynamic_array_reader.h(64)] tag not found
[debug][Application]
Parameters
{
    "_": [],
    "tray": false,
    "hide": false,
    "hidden": false,
    "max": false,
    "maximize": false,
    "maximized": false,
    "min": false,
    "minimize": false,
    "minimized": false,
    "reset-window": false,
    "versioncheck": false,
    "version-check": false,
    "logfile": true,
    "loglevel": "debug",
    "l": "debug",
    "theme": "",
    "launch": "",
    "goto": ""
}
[64214:64222:0515/112955.253034:ERROR:cert_verify_proc_builtin.cc(621)] No net_fetcher for performing AIA chasing.
[64214:64222:0515/112956.135863:ERROR:cert_verify_proc_builtin.cc(621)] No net_fetcher for performing AIA chasing.
[64214:64222:0515/112959.865820:ERROR:cert_verify_proc_builtin.cc(621)] No net_fetcher for performing AIA chasing.
[64214:64222:0515/113007.962638:ERROR:cert_verify_proc_builtin.cc(621)] No net_fetcher for performing AIA chasing.
[64211:64211:0515/113015.082268:ERROR:gl_surface_presentation_helper.cc(260)] GetVSyncParametersIfAvailable() failed for 1 times!
```

**Comment by bastimeyer** [comment]
Did you read https://github.com/streamlink/streamlink-twitch-gui/issues/1008#issuecomment-2107657475 ?
NW.js on Linux has been downgraded to 0.80 on master. STG v2.5.1 with NW.js 0.82 is irrelevant. You didn't even bother to try building from master.

Also give me a break, man... There are tons of other issues with NW.js/Chromium releases, as you can see in the recent releases, the linked issue and the recent changes on master. Tiling window managers are such an edge case, it's barely even relevant.

**Comment by bastimeyer** [comment]
NW2 is a re-implementation of NW.js's JS bindings. NW1 has been removed (or stopped working) in 0.85, hence the removal in this commit. This is also the reason why the exit confirmation dialog started working again, because the app's `close` event is broken in NW1. However, in NW2, the app-window's `blur` event is broken. This has been a pick-your-poison situation for me since at least one and a half years. The issue has been reported by me a long time ago with a couple of bumps, but without any responses.

Considering that the Linux and Windows builds are now stuck on 0.80 and 0.82 respectively, I can have a look at disabling NW2 again.

**Comment by bastimeyer** [comment]
Resolved by 381a9a5cb507ed689b2127ad6afb67127582eef8 / ab45651abefddd9d39d515f184648214ba810c1b
See #1015
